src/app/features/animals/services/animals.service.ts

```diff
@@ -0,0 +1,38 @@
+import {Injectable} from '@angular/core';
+import {Animal} from '../models/animal';
+import {HttpClient} from '@angular/common/http';
+import {tap} from 'rxjs';
+import {AuthorizationService} from '../../../core/services/authorization.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AnimalsService {
+  private _animals: any[] = [];
```

Comment:
Tu by som pouzil `BehaviorSubject`. Tym padom vobec nebudes potrebovat tie `_animals`, kedze hodnoty si budes drzat v tom subject-e. Nasladne sa vies dostat ku aktualne hodnote cez funkciu `getValue` a upravit hotnotu subject-u vies cez `next`. Nasledne si vies v `get animals` z tohto subject-u urobit `Observable` pomocou funkcie `asObservable` a ten pouzivat. Nasledne ked sa ti zmeni hodnota v subject-e tak sa to automaticky prejavy v applikacii.

Reply:
Ako typ pre `_animals` si použil `any`, ale zároveň si pridal model `Animal`. Namiesto typu `any` by si mal všade v `AnimalsService` použiť ako typ interface `Animal`.